### PR TITLE
[Task 1868154] refactor sqlserver-lib to keep the unified module style

### PR DESCRIPTION
**Pull request by shenqianjin**

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/SqlServer.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+    public SqlServer(@Nonnull SqlServerManager manager, @Nonnull com.azure.resourcemanager.sql.models.SqlServer server) {
+        super(new SqlServerEntity(server));
+        this.manager = manager;
+    }
+
+    @Override
+    protected com.azure.resourcemanager.sql.models.SqlServer loadRemote() {
+        try {
+            this.entity().setRemote(manager.sqlServers().getById(this.entity.getId()));
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/SqlServer.java:43]: I think the setRemote shall be handled in AbstractAzureEntityManager

Comment (shenqianjin) [reply azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/SqlServer.java]: updated

File: azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/SqlServer.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        this.manager = manager;
+    }
+
+    @Override
+    protected com.azure.resourcemanager.sql.models.SqlServer loadRemote() {
+        try {
+            return manager.sqlServers().getById(this.entity.getId());
+        } catch (ManagementException ex) {
+            if (HttpStatus.SC_NOT_FOUND == ex.getResponse().getStatusCode()) {
+                this.entity().setRemote(null);
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/SqlServer.java:47]: return null

Comment (shenqianjin) [reply azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/SqlServer.java]: updated.

**Review by andxu — APPROVED** [review]
LGTM

**Pull request merged**